stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -414,15 +414,23 @@ struct Tridiagonal{T,V<:AbstractVector{T}} <: AbstractMatrix{T}
                 "lengths of subdiagonal, diagonal and superdiagonal: ",
                 "($(length(dl)), $(length(d)), $(length(du)))")))
         end
-        new{T,V}(dl, d, du)
+        new{T,V}(dl, d, du, nothing)
     end
     # constructor used in lu!
     function Tridiagonal{T,V}(dl, d, du, du2) where {T,V<:AbstractVector{T}}
         @assert !has_offset_axes(dl, d, du, du2)
+        n = length(d)
+        if (length(dl) != n-1 || length(du) != n-1 || (!isa(du2, Nothing) && length(du2) != n-2))
+            throw(ArgumentError(string("cannot construct Tridiagonal from incompatible ",
+            "lengths of subdiagonal, diagonal and superdiagonal: ",
+            "($(length(dl)), $(length(d)), $(length(du)), $(length(du2)))")))
+        end
         # length checks?
         new{T,V}(dl, d, du, du2)
     end
 end
+Tridiagonal{T, V}(A::Tridiagonal) where {T, V<:AbstractVector{T}} = Tridiagonal(map(x->convert(V, x), (A.dl, A.d, A.du))...)
```

Comment:
This one should probably allocate new memory in any case.
```suggestion
Tridiagonal{T, V}(A::Tridiagonal) where {T, V<:AbstractVector{T}} = Tridiagonal(V(A.dl), V(A.d), V(A.du))
```